apple/internal/apple_library.bzl

```diff
@@ -0,0 +1,309 @@
+                private_hdrs.append(x)
+    if not objc_srcs:
+        fail("""\
+'srcs' must contain Objective-C source files. Use 'swift_library' if this
+target only contains Swift files.""")
+    if not swift_srcs:
+        fail("""\
+'srcs' must contain Swift source files. Use 'objc_library' if this
+target only contains Objective-C files.""")
+
+    name = kwargs.get("name", None)
+    module_name = kwargs.pop("module_name", name)
+    swift_library_name = name + "_swift"
+
+    deps = kwargs.pop("deps", [])
+    objc_deps = []
+    swift_deps = [] + deps
+
+    swift_copts = kwargs.pop("swift_copts", [])
+    swift_copts = swift_copts + [
+        "-Xfrontend",
+        "-enable-objc-interop",
+        "-import-underlying-module",
+    ]
+
+    objc_deps = [":" + swift_library_name]
+
+    # Add Obj-C includes to Swift header search paths
+    repository_name = native.repository_name()
+    includes = kwargs.get("includes", [])
+    for x in includes:
+        include = x if repository_name == "@" else "external/" + repository_name.lstrip("@") + "/" + x
+        swift_copts += [
+            "-Xcc",
+            "-I{}".format(include),
+        ]
+
+    # Generate module map for the underlying Obj-C module
+    textual_hdrs = kwargs.get("textual_hdrs", [])
+    _module_map(
+        name = name + "_objc_module",
+        hdrs = hdrs,
+        module_name = module_name,
+        textual_hdrs = textual_hdrs,
+    )
+
+    objc_module_map = ":" + name + "_objc_module"
+    swiftc_inputs = kwargs.pop("swiftc_inputs", [])
+    swiftc_inputs = swiftc_inputs + hdrs + textual_hdrs + private_hdrs + [objc_module_map]
+
+    swift_copts += [
+        "-Xcc",
+        "-fmodule-map-file=$(execpath {})".format(objc_module_map),
+    ]
+
+    features = kwargs.pop("features", [])
+    swift_features = features + ["swift.no_generated_module_map"]
+
+    swift_library(
+        name = swift_library_name,
+        copts = swift_copts,
+        deps = swift_deps,
+        features = swift_features,
+        generated_header_name = module_name + "-Swift.h",
+        generates_header = True,
+        module_name = module_name,
+        srcs = swift_srcs,
+        swiftc_inputs = swiftc_inputs,
+    )
+
+    umbrella_module_map = name + "_umbrella_module"
+    _module_map(
+        name = umbrella_module_map,
+        deps = [":" + swift_library_name],
+        hdrs = hdrs,
+        module_name = module_name,
+    )
+    objc_deps.append(":" + umbrella_module_map)
+
+    objc_copts = kwargs.pop("objc_copts", [])
+
+    native.objc_library(
+        copts = objc_copts,
+        deps = objc_deps,
+        hdrs = hdrs + [
+            # These aren't headers but here is the only place to declare these
+            # files as the inputs because objc_library doesn't have an
+            # attribute to declare custom inputs.
+            ":" + umbrella_module_map,
+        ],
+        module_map = umbrella_module_map,
```

Comment:
I _think_ you are accounting for the bug I fixed here: https://github.com/buildbuddy-io/rules_xcodeproj/commit/d40173d84764c1256b95a537b2871e010398f18c#diff-9e7ea7f032b7624ef92c4503d20ca20bf16eafd89f6277d71b83dd458972dc3d, but I just wanted to check and make sure.

Reply:
Yes, `objc_library`'s generated module map shouldn't be used.